File: java/Robot.java
/** Robot */

record Point(int x, int y) {}

interface ArtificialIntelligence {
    void move(Point destination);
    Point getPosition();

    default boolean killEnemy(byte command) {
        return checkAsimovLaws(command);
    }

    private boolean checkAsimovLaws(byte command) {
        return command >= 0;
    }
}

class Robot implements ArtificialIntelligence {
    private Point position;
    
    Robot() {
      this.position = new Point(0, 0);
    }

    @Override
    public void move(Point destination) {
        position = destination;
    }

    @Override
    public Point getPosition() {
        return this.position;
    }
}

class Main {
    public static void main(String[] args) {
      
      ArtificialIntelligence aiRobot = new Robot();
      aiRobot.killEnemy((byte) 123);
      
      System.out.println("Works...");
  }
}

File: kotlin/Robot.kt
/** Robot */

data class Point(val x: Int, val y: Int)

interface ArtificialIntelligence {
    fun move(destination: Point)
    fun getPosition(): Point

    fun killEnemy(command: Byte = 0): Boolean {
        return checkAsimovLaws(command)
    }

    private fun checkAsimovLaws(command: Byte): Boolean {
        return command >= 0
    }
}

class Robot : ArtificialIntelligence {
    private var position: Point = Point(0, 0)

    override fun move(destination: Point) {
        position = destination
    }

    override fun getPosition(): Point {
        return position
    }
}

fun main() {
    val aiRobot = Robot()
    aiRobot.killEnemy(123.toByte())

    println("Works...")
}
